src/auro-alaska.js

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable one-var */
+// Copyright (c) 2020 Alaska Airlines. All right reserved. Licensed under the Apache-2.0 license
+// See LICENSE in the project root for license information.
+
+// ---------------------------------------------------------------------
+
+import { html, css } from "lit-element";
+import { classMap } from 'lit-html/directives/class-map';
+import BaseIcon from "./baseIcon";
+
+import as400 from '@alaskaairux/icons/dist/restricted/AS-400_es6.js';
+import as300 from '@alaskaairux/icons/dist/restricted/AS-300_es6.js';
+import as200 from '@alaskaairux/icons/dist/restricted/AS-200_es6.js';
+import as100 from '@alaskaairux/icons/dist/restricted/AS-100_es6.js';
+import asTag400 from '@alaskaairux/icons/dist/restricted/AS-tagline-400_es6.js';
+import asTag300 from '@alaskaairux/icons/dist/restricted/AS-tagline-300_es6.js';
+import asTag200 from '@alaskaairux/icons/dist/restricted/AS-tagline-200_es6.js';
+import asTag100 from '@alaskaairux/icons/dist/restricted/AS-tagline-100_es6.js';
+
+// Import touch detection lib
+// import "focus-visible/dist/focus-visible.min.js";
+import styleCss from "./alaskaStyle-css.js";
+
+// See https://git.io/JJ6SJ for "How to document your components using JSDoc"
+/**
+ * auro-alaska provides users a way to use the Auro Icons by simply passing in the category and name.
+ *
+ * @attr {Boolean} alaska - Set value for default alaska airlines logo
+ * @attr {Boolean} alaskaOfficial - Set value for alaska airlines logo with official tagline
+ * @slot - Hidden from visibility, used for a11y if icon description is needed
+ */
+
+// build the component class
+class AuroAlaska extends BaseIcon {
+  constructor() {
+    super();
+
+    this.privateDefaults();
+  }
+
+  /**
+   * @private internal defaults
+   * @returns {void}
+   */
+  privateDefaults() {
+    this.uri = 'https://unpkg.com/@alaskaairux/icons@latest/dist';
+    this.zero = 0;
+    this.no = 1;
+    this.sm = 107;
+    this.md = 191;
+    this.lg = 527;
+  }
+
+  // function to define props used within the scope of this component
+  static get properties() {
+    return {
+      ...super.properties,
+      alaska: {
+        type: Boolean,
+        reflect: true
+      },
+      alaskaOfficial: {
+        type: Boolean,
+        reflect: true
+      },
+    };
+  }
+
+  /**
+   * @private async function to decide which logo to use
+   * @param {number} iconWidth size of parent icon is in
+   * @returns {object} SVG to be appended to DOM
+   */
+  alaskaLogoDef(iconWidth) {
+    switch (this.alaska) {
```

Comment:
I don't understand how this switch statement works -- the cases have nothing to do with `this.alaska`.

Can this be an if..else if..else if..else instead?

Reply:
There are too many if-else because of the different versions of the logo and the container query. The switch case made the most sense to me.

Reply:
But a switch is for when you're switching between different values of a variable, in this case `this.alaska`. `this.alaska` is a boolean, so it can either be true or false. But you have 3 cases and a default. I don't think this code is doing what you're intending it to do. 

If `this.alaska` is true, it will fall into the first case statement that evaluates to true. If `this.alaska` is false, it will fall into the first case statement that evaluates to false.

Reply:
:+1: